src/theme/SearchBar/index.js

```diff
@@ -5,44 +5,32 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useRef, useCallback, useState, useEffect, version} from "react";
+import React, { useRef, useCallback, useState, useEffect } from "react";
 import classnames from "classnames";
 import { useHistory, useLocation } from "@docusaurus/router";
 import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
 import useVersioning from "@theme/hooks/useVersioning";
 
-const urlMatchesPrefix = (url, prefix) => {
-  if (prefix.endsWith("/")) {
-    throw new Error(`prefix must not end with a /.`);
-  }
-  return url === prefix || url.startsWith(`${prefix}/`);
-};
-
 const determineDocsVersionFromURL = (
   path,
-  basePath,
-  docsBaseRoutePath,
-  versions
+  versionPathRegex
 ) => {
-  // Array of versions that have route prefixes.
-  // The latest version (version[0]) has no route prefix.
-  const routeBasedVersions = ["next", ...versions.slice(1)];
-  for (const version of routeBasedVersions) {
-    if (urlMatchesPrefix(path, `${basePath}${docsBaseRoutePath}/${version}`)) {
-      return version;
-    }
+  if (versionPathRegex) {
+    const matchedVersion = path.match(new RegExp(`^\\/${versionPathRegex}`));
+    //
+    return !!matchedVersion && matchedVersion.length ? matchedVersion[1] : "current";
   }
-  return versions[0];
 };
 
 const Search = props => {
   const initialized = useRef(false);
   const searchBarRef = useRef(null);
   const history = useHistory();
-  const { siteConfig = {} } = useDocusaurusContext();
+  const { siteConfig } = useDocusaurusContext();
+  const { versionPathRegex } = siteConfig?.customFields;
```

Comment:
Do not use the optional chaining operator
```suggestion
  const { versionPathRegex } = siteConfig.customFields;
```